### PR TITLE
[2020] Do a head to get the object size when seeking to end if size i…

### PR DESCRIPTION
**Pull request by JustinKyleJames**
…s unknown.

### CODE REVIEW

File: s3/s3_transport/include/s3_transport.hpp [hunk 1]
```diff
@@ -640,7 +641,19 @@ namespace irods::experimental::io::s3_transport
...
 
                     case std::ios_base::end:
-                        set_file_offset(config_.object_size + _offset);
+
+                        if (existing_object_size == config::UNKNOWN_OBJECT_SIZE) {
+                            // do a stat to get object size
+                            object_s3_status object_status = object_s3_status::DOES_NOT_EXIST;
+                            irods::error ret = get_object_s3_status(object_key_, bucket_context_, existing_object_size, object_status);
+                            if (!ret.ok() || object_status == object_s3_status::DOES_NOT_EXIST) {
+                                 rodsLog(LOG_ERROR, "%s:%d (%s) [[%lu]] seek failed because object size is unknown and HEAD failed",
```
Comment (trel) [inline s3/s3_transport/include/s3_transport.hpp:650]: indented one-too-many spaces

File: s3/s3_transport/unit_tests/src/test_s3_transport.cpp [hunk 2]
```diff
@@ -1041,6 +1089,30 @@ TEST_CASE("s3_transport_readwrite_thread", "[rw][thread]")
...
+    std::string object_prefix = "dir1/dir2/";
+
+    SECTION("seek end small file")
+    {
+        filename = "small_file";
+        test_seek_end(bucket_name, filename, object_prefix, keyfile);
+
+    }
+
+    SECTION("seek end medium file")
```
Comment (trel) [inline s3/s3_transport/unit_tests/src/test_s3_transport.cpp:1108]: are these small and medium files already created/managed by the test framework?

Comment (JustinKyleJames) [reply s3/s3_transport/unit_tests/src/test_s3_transport.cpp:1108]: There is a script that can be run to create them.

File: s3/s3_transport/include/s3_transport.hpp (hunk 1 above)
Comment (JustinKyleJames) [reply s3/s3_transport/include/s3_transport.hpp]: fixed

File: s3/s3_transport/unit_tests/src/test_s3_transport.cpp (hunk 2 above)
Comment (JustinKyleJames) [reply s3/s3_transport/unit_tests/src/test_s3_transport.cpp:1108]: My mistake.  There is no README for the unit tests but the script is there.

Comment (trel) [reply s3/s3_transport/unit_tests/src/test_s3_transport.cpp:1108]: got it - just checking as it seemed to reference something I couldn't see.

**Comment by trel** [comment]
Does @d-w-moore need to see this before it's merged?

Otherwise, please # it.

**Comment by JustinKyleJames** [comment]
> Does @d-w-moore need to see this before it's merged?
> 
> Otherwise, please # it.

He doesn't.  I used his script as a manual test.  Pound was added.

**Comment by d-w-moore** [comment]
I will nevertheless be testing it for the parallel get( ).  Quite sure it it will pass....

**Pull request merged**